### PR TITLE
Fix indexing same block with trace_filter and trace_block

### PR DESCRIPTION
**Pull request by Uxio0**
- Previously when indexing in mixed `trace_filter`/`trace_block `mode last block for `trace_filter `was also indexed by `trace_block`

Also:
- Add more debug log messages on the indexers
- Update some function docstrings

### CODE REVIEW

File: safe_transaction_service/history/indexers/internal_tx_indexer.py [hunk 1]
```diff
@@ -239,22 +233,22 @@ def _get_internal_txs_to_decode(
         self, tx_hashes: Sequence[str]
     ) -> Generator[InternalTxDecoded, None, None]:
         """
-        Use generator to be more RAM friendly
+        Retrieve relevant `InternalTxs` and if possible decode them to return `InternalTxsDecoded`
+
+        Every `InternalTx` must be retrieved again as `bulk_create` with `ignore_conflicts=True`
```
Comment (moisses89) [inline safe_transaction_service/history/indexers/internal_tx_indexer.py:238]: I spend sometime trying to understand why you are mentioning here the bulk_create :smiling_face_with_tear: 
Maybe could be interesting to add that this functions is returning a generator to be used in a bulk_create_from_generator.

File: safe_transaction_service/history/indexers/internal_tx_indexer.py [hunk 2]
```diff
@@ -141,7 +131,7 @@ def find_relevant_elements(
             )
             relevant_elements.update(
                 self._find_relevant_elements_using_trace_block(
-                    addresses, trace_block_number, to_block_number
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply safe_transaction_service/history/indexers/internal_tx_indexer.py:144]: Agree, changed

File: safe_transaction_service/history/indexers/internal_tx_indexer.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/indexers/internal_tx_indexer.py]: Agree, I moved the comment after `InternalTx.objects.bulk_create_from_generator`

**Review by moisses89 — APPROVED** [review]
Looks good to me!

**Pull request merged**